include/config.js

```diff
@@ -113,585 +93,287 @@ var CONFIG_MODULE_NAME  = 'config.js';
+/** 
+ * Retrieve the base configuration
  */
-var BASE_CONFIG = {
-
-    //The name of the site.
-	siteName: 'pencilblue',
-
-    //The root of the site.  This host part should ALWAYS match the value of
-    //the siteIP
-	siteRoot: 'http://localhost:8080',
-
-    //The hostname or IP address that the web server instance will bind to
-	siteIP:   '0.0.0.0',
-
-    //The primary port to listen for traffic on.  Some environment such as
-    //heroku force you to use whatever port they have available.  In such cases
-    //the port is passed as an environment variable.
-	sitePort: process.env.port || process.env.PORT || 8080,
-
-    //the absolute file path to the directory where installation lives
-	docRoot:  Configuration.DOCUMENT_ROOT,
-
-    //provides a configuration for connecting to persistent storage.  The
-    //default configuration is meant for mongodb.
-	db: {
-        type:'mongo',
-		servers: [
-          '127.0.0.1:27017'
-        ],
-
-        //the name of the default DB for the system
-        name: 'pencil_blue',
-
-        //http://docs.mongodb.org/manual/core/write-concern/
-        writeConcern: 1,
-
-        //PB provides the ability to log queries.  This is handy during
-        //development to see how many trips to the DB a single request is
-        //making.  The queries log at level "info".
-        query_logging: false,
-
-        //http://mongodb.github.io/node-mongodb-native/api-generated/db.html#authenticate
-        authentication: {
-            un: null,
-            pw: null,
-            options: {
-                //authMechanism: "MONGODB-CR"|"GSSAPI"|"PLAIN", //Defaults to MONGODB-CR
-                //authdb: "db name here", //Defaults to the db attempted to be connected to
-                //authSource: "db name here", //Defaults to value of authdb
-            }
+Configuration.getBaseConfig = function(multisite) {
+    return util.clone({
+
+        //The name of the site.
+        siteName: 'pencilblue',
+
+        //The root of the site.  This host part should ALWAYS match the value of
+        //the siteIP
+        siteRoot: 'http://localhost:8080',
+
+        //The hostname or IP address that the web server instance will bind to
+        siteIP:   '0.0.0.0',
+
+        //The primary port to listen for traffic on.  Some environment such as
+        //heroku force you to use whatever port they have available.  In such cases
+        //the port is passed as an environment variable.
+        sitePort: process.env.port || process.env.PORT || 8080,
+
+        //the absolute file path to the directory where installation lives
+        docRoot:  Configuration.DOCUMENT_ROOT,
+
+        //enables/disables multiple sites in a single pencilblue instance (multitenancy)
+        multisite: false,
+
+        // When multisite is true, this is the hostname that will resolve to the global namespace.
+        // Only Admin routes will be activated for this hostname.
+        globalRoot: 'http://global.localhost:8080',
```

Comment:
Multi-site is going to get more complex over time.  We need to add a block ```mutisite``` and it be an object rather than multiple top level properties.  This gives us a path forward rather than refactoring it at a later time with breaking changes.